### PR TITLE
Refactor whole page

### PR DESCRIPTION
**Pull request by marcelgerber**
I noticed a few mistakes here and there, so this PR includes many fixes and little improvements for the overall script part of the page.

It doesn't change UI, UX or behaviour, and it's definitely not urgent.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -421,7 +422,7 @@ <h4 data-i18n="index.page.requirements.updates.header">Frequent Updates</h4>
         unsupported.os = "Debian 7";
         unsupported.i18nKey = "debian-7";
         unsupported.url = "https://github.com/adobe/brackets/issues/4816";
-        unsupported.default = "Unfortunately, we currently don't support Debian 7. <a href='" + unsupported.url + "'>Take a look at the GitHub issue</a> for further information.";
+        unsupported.defaultText = "Unfortunately, we currently don't support Debian 7. <a href='" + unsupported.url + "'>Take a look at the GitHub issue</a> for further information.";
```
Comment (Mark-Simulacrum) [inline index.html]: Might consider doing "don't currently" instead of "currently don't" to better match the "don't support Windows XP" pattern.

File: index.html [hunk 2]
```diff
@@ -429,19 +430,20 @@ <h4 data-i18n="index.page.requirements.updates.header">Frequent Updates</h4>
     }
 
     if (unsupported.os) {
-        if (!unsupported.default) {
-            unsupported.default = "Unfortunately, we don't support " + unsupported.os + " any longer. <a href='" + unsupported.url + "'>Take a look at the Requirements.</a>";
+        if (!unsupported.defaultText) {
+            unsupported.defaultText = "Unfortunately, we don't support " + unsupported.os + " any longer. <a href='" + unsupported.url + "'>Take a look at the Requirements.</a>";
```
Comment (Mark-Simulacrum) [inline index.html]: Linking to the requirements doesn't make much sense. The requirements don't list and reasons why they are the requirements - and as such, don't help someone understand why support for and OS was dropped. I would expect that any such dropping of support would have a blog post (similar to the Windows XP blog post), and would expect a link to that instead of the requirements.

Perhaps linking to the requirements is a good idea both here and in the Windows XP unsupported section.

File: index.html [hunk 3]
```diff
@@ -487,12 +489,7 @@ <h4 data-i18n="index.page.requirements.updates.header">Frequent Updates</h4>
                         url = "https://github.com/adobe/brackets/releases/download/" + tag + "/Brackets." + buildName.split(" ").join(".") + ext;
                     }
 
-                    ///if(OS=="WIN") url = "http://bit.ly/QrVHNc"; // (Temporary) TRACKING WIN
-                    ///else if(OS=="OSX") url = "http://bit.ly/1hQGEsh"; // (Temporary) TRACKING OSX
-
-                    var version_label = buildName.match(/(\d+.*$)/)[1];
-                    //if (OS != "UNKNOWN") version_label += " (" + OS + ")";
-                    $("#download-brackets-version").text(version_label);
+                    $("#download-brackets-version").text(buildNum);
```
Comment (Mark-Simulacrum) [inline index.html:498]: Is this change intended? The functionality was changed from `version_label` being a match of `\d+.*$` to `buildNum`'s `[\d.]+` match, which are different things.

File: index.html [hunk 4]
```diff
@@ -504,12 +501,12 @@ <h4 data-i18n="index.page.requirements.updates.header">Frequent Updates</h4>
...
 
             // update features (limit 5)
-            var feature_content = "<h2>" + i18n.t("index.page.content.new-features.new-features-in", { defaultValue: "New in " + build.versionString, sprintNumber: build.versionString }) + "</h2>";
-            build.newFeatures.slice(0, Math.min(build.newFeatures.length, 5)).forEach(function (feature) {
-                feature_content += "<h3>" + $("<div/>").text(feature.name).html() + "</h3>";
-                feature_content += "<p>" + $("<div/>").text(feature.description).html() + "</p>";
+            var feature_content = '<h2>' + i18n.t("index.page.content.new-features.new-features-in", { defaultValue: "New in " + build.versionString, versionString: build.versionString }) + '</h2>';
+            build.newFeatures.slice(0, 5).forEach(function (feature) {
+                feature_content += '<h3>' + $("<div/>").text(feature.name).html() + '</h3>';
+                feature_content += '<p>' + $("<div/>").text(feature.description).html() + '</p>';
```
Comment (Mark-Simulacrum) [inline index.html:513]: Why the change from double quote (`"`) to single quote (`'`). I see why on line [509](https://github.com/adobe/brackets.io/pull/139/files#diff-eacf331f0ffc35d4b482f1d15a887d3bR509), but I don't see why here. (My assumption on 500 is that you were trying to use double quotes for the attributes).

File: index.html [hunk 5]
```diff
@@ -518,68 +515,72 @@ <h4 data-i18n="index.page.requirements.updates.header">Frequent Updates</h4>
         var elem = document.getElementById(elementID);
         elem.href = _url;
         elem.onclick = function (e) {
```
Comment (Mark-Simulacrum) [inline index.html]: Why don't we use a jQuery event handler here?

File: index.html [hunk 6]
```diff
@@ -518,68 +515,72 @@ <h4 data-i18n="index.page.requirements.updates.header">Frequent Updates</h4>
         var elem = document.getElementById(elementID);
```
Comment (Mark-Simulacrum) [inline index.html]: Why is this using non-jQuery element getter? Is this for performance reasons?

File: index.html [hunk 7]
```diff
@@ -518,68 +515,72 @@ <h4 data-i18n="index.page.requirements.updates.header">Frequent Updates</h4>
...
         elem.href = _url;
         elem.onclick = function (e) {
-            e.preventDefault();
-            var downloadStarted = false;
-            setTimeout(function () { if (!downloadStarted) window.location.href = _url; }, 1200); // fallback
-            _gaq.push(tagObj, function () {
-                downloadStarted = true;
-                setTimeout(function () { window.location.href = _url; }, 1000); // short delay just in case it doesn't get logged
-            });
+            if (e.which === 2) { // middle button => current tab will stay open => will be logged in any case
```
Comment (Mark-Simulacrum) [inline index.html]: According to MDN, `e.which` for the `click` event doesn't exist. Perhaps you meant `e.button` (where it would be `=== 1`, according to MDN.)

File: index.html (hunk 3 above)
Comment (marcelgerber) [inline index.html:498]: Yeah, it is intended as until now, all releases were named using the pattern `Release n.m`, which makes both RegExps match the same.

File: index.html (hunk 7 above)
Comment (marcelgerber) [inline index.html]: Where did you look? According to https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/which, every MouseEvent has a `.which` property and it's more standardised than `.button`.

File: index.html (hunk 4 above)
Comment (marcelgerber) [inline index.html:513]: I tried to streamline quotes a little, so every HTML literal uses `'` as outer quotes.

File: index.html (hunk 7 above)
Comment (Mark-Simulacrum) [inline index.html]: From what I can see, `MouseEvent.which` isn't a standard (and is not on a standards track), as said in the box at the top of the page you linked (https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/which).

> Non-standard
> This feature is non-standard and is not on a standards track. Do not use it on production sites facing the Web: it will not work for every user. There may also be large incompatibilities between implementations and the behavior may change in the future.

On the other hand, `MouseEvent.button` is standardized and appears to be supported almost the same as `.which`. (Source: https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/button)

File: index.html [hunk 8]
```diff
@@ -367,8 +367,10 @@ <h4 data-i18n="index.page.requirements.updates.header">Frequent Updates</h4>
 </script>
 
 <script>
-    var ext = "";
-    var OS = "UNKNOWN";
+    // Detect Operating System and required file extension
+    var ext = "",
+        OS = "UNKNOWN";
+
     if (/Windows|Win32|WOW64|Win64/.test(navigator.userAgent)) {
```
Comment (le717) [inline index.html]: If you wanted to be more generic and possibly catch any changes (if there are ever changes), you could use

`if (w[in|ow]/i.test(navigator.userAgent)) {`

In fact, it'd probably be a good idea to make these checks case-insensitive.

File: index.html [hunk 9]
```diff
@@ -410,10 +411,10 @@ <h4 data-i18n="index.page.requirements.updates.header">Frequent Updates</h4>
...
         unsupported.i18nKey = "win-xp";
         unsupported.url = "http://blog.brackets.io/2014/04/18/windows-xp-support/";
-        unsupported.default = "Unfortunately, we don't support Windows XP any longer. <a href='" + unsupported.url + "'>Read our blog post</a> for further information.";
-    } else if (osMatch = navigator.userAgent.match(/Mac OS X 10[._](\d+)([._](\d+))?\D/i)) { // Mac OS X 10.*
-        osMatch = parseFloat(osMatch[1] + "." + (osMatch[3] || 0));
-        if (osMatch < 6.8 && osMatch !== 6.0) { // Firefox reports only the OS X major version (10.6), so we have to check for 6.0
+        unsupported.defaultText = "Unfortunately, we don't support Windows XP any longer. <a href='" + unsupported.url + "'>Read our blog post</a> for further information.";
+    } else if (osMatch = navigator.userAgent.match(/Mac OS X 10[._](\d+)(?:[._](\d+))?\D/i)) { // Mac OS X 10.*
+        osMatch = parseFloat(osMatch[1] + "." + (osMatch[2] || 0));
+        if (osMatch < 6.8 && osMatch !== 6.0) { // Firefox only reports the OS X major version (10.6), so we have to check it's not 6.0
```
Comment (le717) [inline index.html]: IDK, to me if feels like this area needs some whitespace, and change this long comments to sit above the condition, not inline. IMO, inline comments should only be used for two or three word comments.

**Comment by le717** [comment]
Line 410 personal nit (so totally optional): `if (/NT ?5\.\d/.test(navigator.userAgent)) { // Windows XP` Use `\s` in the regex instead of actual space. :)

File: index.html (hunk 1 above)
Comment (le717) [inline index.html]: @Mark-Simulacrum I see your logic, but as a native English speaker and one who strives to use proper English, especially in website design, "don't currently" just does not work. The original "We currently [do not] support Debian 7" was better, in both understanding and grammar (though in personal web design, I eliminate contractions completely).

**Comment by le717** [comment]
@MarcelGerber I know you said

> It doesn't change UI, UX or behaviour

But I can't help but wonder if you would make this PR automatically generate the E4B download link like it should be. :)

**Comment by le717** [comment]
Also, maybe this is not the PR for it, but with your changes this one page loads _13_ scripts (one is the Source Sans Pro font). Is there a way any of scripts can be concatenated together (and possibly made `async`) in order to reduce the amount of requests made?

File: index.html (hunk 8 above)
Comment (marcelgerber) [inline index.html]: There _shouldn't_ be any changes to UserAgent strings, at least not in common browsers, but good point about the case-insensitivity. I thought that too a while ago.

**Comment by marcelgerber** [comment]
@le717 Addressed most of your review comments. Thanks!
Re concatenation/minification: That would obviously improve load time a lot, but the problem is that we'd need a build step to do that. Not that that's impossible to do, but I have little experience with build tools and thus, I'm not the perfect person for this.

**Comment by le717** [comment]
> Re concatenation/minification: That would obviously improve load time a lot, but the problem is that we'd need a build step to do that. Not that that's impossible to do, but I have little experience with build tools and thus, I'm not the perfect person for this.

One simple way is to make a .js file and move as much code as possible (logically organized, of course) into it then link that one script instead of the many scripts. :wink:

Since this is running on GitHub Pages, if the CSS was converted to Sass in the future, we could have Jekyll auto-minify that (unfortunately the same does not go for JS unless it was ported to CoffeeScript (ehhhhhhhhh)).

> Addressed most of your review comments. Thanks!

I actually did more of a quick glance. It was near the end of class and I was bored, so while browsing GitHub I saw this PR, read over it, and commented when I got home. ;P

**Comment by ficristo** [comment]
@MarcelGerber do you still want this?
